airflow/jobs/job.py

```diff
@@ -156,7 +152,10 @@ def kill(self, session: Session = NEW_SESSION) -> NoReturn:
     def on_kill(self):
         """Will be called when an external kill command is received."""
 
-    def heartbeat(self, only_if_necessary: bool = False) -> None:
+    @provide_session
+    def heartbeat(
+        self, heartbeat_callback: Callable[[SASession], None], session: Session = NEW_SESSION
```

Comment:
```suggestion
        self, heartbeat_callback: Callable[[Session], None], session: Session = NEW_SESSION
```
Won't `Session` from orm.session serve?

Reply:
I copied it from other places where Session has been renamed to SASession, but it is likely not needed here. I will update it.